src/codecs/webp/extended.rs

```diff
@@ -326,10 +326,27 @@ impl ExtendedImage {
     pub(crate) fn fill_buf(&self, buf: &mut [u8]) {
         match &self.image {
             // will always have at least one frame
-            ExtendedImageData::Animation { frames, .. } => &frames[0].image,
-            ExtendedImageData::Static(image) => image,
+            ExtendedImageData::Animation { frames, anim_info } => {
+                let first_frame = &frames[0];
+                let (canvas_width, canvas_height) = self.dimensions();
+                if canvas_width == first_frame.width && canvas_height == first_frame.height {
+                    first_frame.image.fill_buf(buf);
+                } else {
```

Comment:
I'm slightly nervous that `first_frame.width` or `first_frame.height` could be larger than the size of the canvas. Do you know what would happen in that case?

Reply:
Good point. If the frame's width or height are larger than the canvas size, or even if they are smaller, but they have an x or y offset + width/height, that gets larger than the canvas width/height, than the code will panic on [extended.rs#L268](https://github.com/image-rs/image/blob/master/src/codecs/webp/extended.rs#L268), because of how Index is implemented for ImageBuffer.

We can make an additional check if the calculated index is within a canvas size [extended.rs#266](https://github.com/image-rs/image/blob/master/src/codecs/webp/extended.rs#L266) and if some x or y is out of bounds, then break the corresponding loop. Some sort of a crop the frame if it is outside the canvas.

Reply:
If it isn't too much trouble, could you implement that cropping logic now? Otherwise it can wait for a followup PR

Reply:
I created #1960, you can assign it to me. I'll implement it in a separate PR if you don't mind.